proj4/src/test/scala/geotrellis/proj4/io/wkt/WKTParserTest.scala

```diff
@@ -658,4 +659,52 @@ class WKTParserTest extends FunSpec {
 
     assert(strip == single)
   }
+
+  it("Should parse WKT string with proj4 extension") {
+    val strip = WKTParser("""|PROJCS["WGS 84 / Pseudo-Mercator",
+                             |GEOGCS["WGS 84",
+                             |DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],AUTHORITY["EPSG","6326"]],
+                             |PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],
+                             |UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]],
+                             |AUTHORITY["EPSG","4326"]],
+                             |PROJECTION["Mercator_1SP"],
+                             |PARAMETER["central_meridian",0],
+                             |PARAMETER["scale_factor",1],
+                             |PARAMETER["false_easting",0],
+                             |PARAMETER["false_northing",0],
+                             |UNIT["metre",1,AUTHORITY["EPSG","9001"]],
+                             |AXIS["Easting",EAST],
+                             |AXIS["Northing",NORTH],
+                             |EXTENSION["PROJ4","+proj=merc +a=6378137 +b=6378137 +lat_ts=0 +lon_0=0 +x_0=0 +y_0=0 +k=1 +units=m +nadgrids=@null +wktext +no_defs"],
+                             |AUTHORITY["EPSG","3857"]]""".stripMargin)
+
+    // single line equivalent
+    val single = WKTParser("""PROJCS["WGS 84 / Pseudo-Mercator",GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]],AUTHORITY["EPSG","4326"]],PROJECTION["Mercator_1SP"],PARAMETER["central_meridian",0],PARAMETER["scale_factor",1],PARAMETER["false_easting",0],PARAMETER["false_northing",0],UNIT["metre",1,AUTHORITY["EPSG","9001"]],AXIS["Easting",EAST],AXIS["Northing",NORTH],EXTENSION["PROJ4","+proj=merc +a=6378137 +b=6378137 +lat_ts=0 +lon_0=0 +x_0=0 +y_0=0 +k=1 +units=m +nadgrids=@null +wktext +no_defs"],AUTHORITY["EPSG","3857"]]""")
+
+    assert(strip == single)
```

Comment:
Hmm what am I missing here? This test is just checking to see if two strings that are the same are parsing the same, but not specifically checking whether or not the extension was properly parsed?

Reply:
@CloudNiner yea just checks the multistring parsing :D I can make it more smart, thanks :+1: